src/Repositories/TestRepository.php

```diff
@@ -140,6 +142,14 @@ public function set(TestCaseFactory $test): void
             throw new TestAlreadyExist($test->filename, $test->description);
         }
 
+        if ($test->dependent === false && count($test->datasets) === 0) {
```

Comment:
Can't this just be `count($test->datasets) === 0`?

Reply:
Unfortunately not, because tests with dependencies can/will have function arguments, but no dataset so they will fail the test. I.e., we test for datasets & dependencies; if neither exist, then you have (presumably) not supplied any arguments to the test.

Reply:
Thoughts on moving this `if` logic into the `TestCaseFactory`? Something like:

```php
if ($test->doesNotSupportArguments())
```

Reply:
Yeah, I like that idea. I will refactor into this method

Reply:
Updated @lukeraymonddowning, @nunomaduro :)